### PR TITLE
feat: add proper constructor support

### PR DESCRIPTION
**Pull request by azjezz**
before this change, the http-client example would panic in PHP 8.1 and 8.2

### CODE REVIEW

File: phper/src/classes.rs [hunk 1]
```diff
@@ -85,21 +90,40 @@ impl<T: Send + 'static> StatefulClass<T> {
...
+    pub fn add_method<F, R>(
+        &mut self, name: impl Into<String>, vis: Visibility, handler: F, arguments: Vec<Argument>,
+    ) where
+        F: Fn(&mut StatefulObj<T>, &mut [ZVal]) -> R + Send + Sync + 'static,
+        R: Into<ZVal> + 'static,
+    {
+        let name: String = name.into();
+        if name == CONSTRUCTOR_FUNCTION_NAME {
+            self.add_constructor(vis, handler, arguments);
+        } else {
```
Comment (azjezz) [inline phper/src/classes.rs:116]: to keep backward compatibility.

**Comment by sashaaro** [comment]
I am curious can we create shared http `reqwest::blocking` client once for php-fpm. which will be alive, always reuse and do work in backround for keep http2 multiplexing connection. As I know `reqwest::blocking` support it.
Can we implement shared resources between request through sapi

**Comment by azjezz** [comment]
@sashaaro haven't tried, but definitely possible, will make an example once #65 is done.

**Comment by azjezz** [comment]
also to note, I'm planning to building something to support async handlers, through PHP Fibers, but currently not a priority and might take a while

**Comment by sashaaro** [comment]
I create https://github.com/sashaaro/reqwest-php-extension  but still don't know how to create `reqwest::blocking::Client` once.
Would be like to hear you thoughts and assistance. Maybe I totally wrong and it is not worth it

**Comment by jmjoy** [comment]
In the php-fpm mode, if you want to share resources among various worker processes, you need to put the resources in the master process or a new sub process, and then let the worker process interact with the resources through the IPC, which is somewhat complicated.

**Comment by jmjoy** [comment]
@azjezz You need to pass the CI before I can review it.

**Comment by azjezz** [comment]
@jmjoy the CI issue is unrelated to the PR, not sure why clippy keeps showing this same issue in all the PRs, but not the main branch ^^" fixing the issues results in a build failure as you can see.

**Comment by jmjoy** [comment]
I cannot merge your PR when CI fails. Please understand.

**Comment by azjezz** [comment]
i know ^^ i was checking if you knew what the clippy issue is.

**Comment by jmjoy** [comment]
But, now the CI failed isn't because clippy but test.

**Comment by azjezz** [comment]
reverted the clippy changed, and disabled it, will see how this goes :)

**Comment by sashaaro** [comment]
> In the php-fpm mode, if you want to share resources among various worker processes, you need to put the resources in the master process or a new sub process, and then let the worker process interact with the resources through the IPC, which is somewhat complicated.

@jmjoy probably for IPC I can use socket file descriptor,
but as I understand share resources can work without IPC in PHP zts mode.
Support PHP zts mode in your plan to implement? I would be happy to see it

**Pull request closed without merging**